src/main/java/uk/gov/justice/probation/courtcaseservice/controller/model/CourtCaseRequest.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.justice.probation.courtcaseservice.controller.model;
+
+import lombok.Builder;
+import lombok.Data;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.AddressPropertiesEntity;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.CourtCaseEntity;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.ImmutableOffenceEntity;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Data
+@Builder
+public class CourtCaseRequest {
+    private final String caseId;
+    private final String caseNo;
+    private final String courtCode;
+    private final String courtRoom;
+    private final LocalDateTime sessionStartTime;
+    private final String probationStatus;
+    private final LocalDate previouslyKnownTerminationDate;
+    private final Boolean suspendedSentenceOrder;
+    private final Boolean breach;
+    private final List<OffenceRequest> offences;
+    private final String defendantName;
+    private final AddressRequest defendantAddress;
+    private final LocalDate defendantDob;
+    private final String defendantSex;
+    private final String crn;
+    private final String pnc;
+    private final String cro;
+    private final String listNo;
+    private final String nationality1;
+    private final String nationality2;
+
+    public CourtCaseEntity asEntity() {
+        final List<ImmutableOffenceEntity> offences = IntStream.range(0, Optional.ofNullable(getOffences())
+                .map(List::size)
+                .orElse(0)
+        )
+                .mapToObj(i -> {
+                    var offence = getOffences().get(i);
+                    return ImmutableOffenceEntity.builder()
```

Comment:
An observation I've been meaning to make for a while. We are supplied with an offence sequence number from the incoming payload which we disregard. This is fine, though

Reply:
Ah was not aware of that, I've removed that from the `OffenceRequest` pojo so this might be a breaking change then...